entrypoint.sh

```
@@ -51,6 +51,19 @@ case ${RUNNER_SCOPE} in
     ;;
 esac
 
+# Loading the files from the mounted directory
+if [ -d ${CONFIGURED_ACTIONS_RUNNER_FILES_DIR} ]; then
```

i dont think we want to check for the existence of the dir but rather the env var, this doesnt seem safe

---

I do not really know the threat from the checking the envvar for the directory existence. Do you suggest using the "do and ignore errors" approach?

---

i think  checking for both the env var and the directory is fine, but only the directory with a default value can add a weird edge case. if its not set then i think you can short circuit and do nothing regardless of the directory presence

---

How about the following: `if [ -d ${CONFIGURED_ACTIONS_RUNNER_FILES_DIR:=/actions-runner-files} ]; then` ?
https://stackoverflow.com/a/2013589

---

In this usage its a bit hard to read and a bit smelly to check on something that mutates (`:=`)

---

If not `:=` then for every such the check `:-` is needed

---

It was changed